core/src/SYCL/Kokkos_SYCL_Parallel_Range.hpp

```diff
@@ -130,4 +132,155 @@ class Kokkos::Impl::ParallelFor<FunctorType, ExecPolicy,
       : m_functor(arg_functor), m_policy(arg_policy) {}
 };
 
+// ParallelFor
+template <class FunctorType, class... Traits>
+class Kokkos::Impl::ParallelFor<FunctorType, Kokkos::MDRangePolicy<Traits...>,
+                                Kokkos::Experimental::SYCL> {
+ public:
+  using Policy = Kokkos::MDRangePolicy<Traits...>;
+
+ private:
+  using array_index_type = typename Policy::array_index_type;
+  using index_type       = typename Policy::index_type;
+  using LaunchBounds     = typename Policy::launch_bounds;
+  using WorkTag          = typename Policy::work_tag;
+
+  const FunctorType m_functor;
+  const Policy m_policy;
+
+  ParallelFor()        = delete;
+  ParallelFor& operator=(ParallelFor const&) = delete;
```

Comment:
We should declare the copy/move constructor and move assignment operator as deleted and the destructor as defaulted (Rule of 5).